packages/ui/react-ui-editor/src/extensions/modes.ts

```diff
@@ -9,28 +9,30 @@ import { vscodeKeymap } from '@replit/codemirror-vscode-keymap';
 
 export const focusEvent = 'focus.container';
 
-export type EditorMode = 'default' | 'vim' | 'vscode' | undefined;
+export const EditorViewModes = ['live-preview', 'read-only', 'source'] as const;
```

Comment:
re-use definition in plugin.
why use this form rather than EditorInputMode.
if good reason, then fix other (consistency is best).